backend/src/api/explorer/channels.api.ts

```diff
@@ -288,21 +288,36 @@ class ChannelsApi {
 
       const channels: any[] = []
       for (const row of allChannels) {
-        const activeChannelsStats: any = await nodesApi.$getActiveChannelsStats(row.public_key);
-        channels.push({
-          status: row.status,
-          closing_reason: row.closing_reason,
-          capacity: row.capacity ?? 0,
-          short_id: row.short_id,
-          id: row.id,
-          fee_rate: row.node1_fee_rate ?? row.node2_fee_rate ?? 0,
-          node: {
-            alias: row.alias.length > 0 ? row.alias : row.public_key.slice(0, 20),
-            public_key: row.public_key,
-            channels: activeChannelsStats.active_channel_count ?? 0,
-            capacity: activeChannelsStats.capacity ?? 0,
-          }
-        });
+        let channel;
+        if (index >= 0) {
+          const activeChannelsStats: any = await nodesApi.$getActiveChannelsStats(row.public_key);
+          channel = {
+            status: row.status,
+            closing_reason: row.closing_reason,
+            capacity: row.capacity ?? 0,
+            short_id: row.short_id,
+            id: row.id,
+            fee_rate: row.node1_fee_rate ?? row.node2_fee_rate ?? 0,
+            node: {
+              alias: row.alias.length > 0 ? row.alias : row.public_key.slice(0, 20),
+              public_key: row.public_key,
+              channels: activeChannelsStats.active_channel_count ?? 0,
+              capacity: activeChannelsStats.capacity ?? 0,
+            }
+          };
+        } else if (index === -1) {
```

Comment:
Maybe just an `else` here instead of `else if ` since channels will push undefined if index is -2 or something else .. maybe not important but.

Reply:
The correct fix is to return an appropriate http error code if the query parameter is invalid. I will add this test in the route.

Reply:
Added query parameter test in https://github.com/mempool/mempool/pull/2381/commits/43cc9499b1dbfdd18ca0c12a1e4e07aa2e9462ef